modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1373 @@
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenThrow(InternalErrorException.class);
+        when(userManager.createMe(anyObject(), anyObject())).thenReturn(null);
+
+        //Assertions
+        Assert.assertThrows(InternalErrorException.class, () -> {
+            meResourceManager.create(scimObjectString, userManager, attributes,
+                    excludeAttributes);
+        });
+
+    }
+
+    @DataProvider(name = "dataForTestCreateExceptions")
+    public Object[][] dataToTestCreatExceptions()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        String scimObjectString = getNewUserSCIMObjectString();
+        User user = getNewUser();
+
+        return new Object[][]{
+                {scimObjectString, "userName", null, user}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateExceptions")
+    public void testCreateBadRequestException(String scimObjectString, String attributes,
+                                              String excludeAttributes, Object objectUser)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        //user variable is not used here
+        User user = (User) objectUser;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn("https://localhost:9443/scim2/Me");
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenThrow(BadRequestException.class);
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(BadRequestException.class);
+
+        //Assertions
+        Assert.assertThrows(BadRequestException.class, () -> {
```

Comment:
You can add expected exceptions to @Test annotation in the method get rid of the assertion here

Reply:
Fixed in commit b3385c2ce740925ccf69bcb3792cd019b0c49b30